demos/mqtt/mqtt_demo_connection_sharing/mqtt_demo_connection_sharing.c

```diff
@@ -0,0 +1,1532 @@
+         * callback. This demo uses QOS1 in Subscribe, therefore, the Publish messages received
+         * from the broker will have QOS1. */
+        LogInfo( ( "Subscribing to the MQTT topic %.*s.",
+                   topicFilterLength,
+                   pTopicFilter ) );
+
+        returnStatus = subscribeToTopic( pContext,
+                                         pTopicFilter,
+                                         topicFilterLength );
+    }
+
+    if( returnStatus != EXIT_SUCCESS )
+    {
+        /* Remove the registered callback for the temperature topic filter as
+        * the subscription operation for the topic filter did not succeed. */
+        ( void ) SubscriptionManager_RemoveCallback( pTopicFilter,
+                                                     topicFilterLength );
+    }
+
+    return returnStatus;
+}
+
+/*-----------------------------------------------------------*/
+
+static int unsubscribeFromTopic( MQTTContext_t * pMqttContext,
+                                 const char * pTopicFilter,
+                                 uint16_t topicFilterLength )
+{
+    int returnStatus = EXIT_SUCCESS;
+    MQTTStatus_t mqttStatus;
+    MQTTSubscribeInfo_t pSubscriptionList[ 1 ];
+
+    assert( pMqttContext != NULL );
+
+    /* Start with everything at 0. */
+    ( void ) memset( ( void * ) pSubscriptionList, 0x00, sizeof( pSubscriptionList ) );
+
+    pSubscriptionList[ 0 ].pTopicFilter = pTopicFilter;
+    pSubscriptionList[ 0 ].topicFilterLength = topicFilterLength;
+
+    /* Generate packet identifier for the UNSUBSCRIBE packet. */
+    globalUnsubscribePacketIdentifier = MQTT_GetPacketId( pMqttContext );
+
+    /* Send UNSUBSCRIBE packet. */
+    mqttStatus = MQTT_Unsubscribe( pMqttContext,
+                                   pSubscriptionList,
+                                   sizeof( pSubscriptionList ) / sizeof( MQTTSubscribeInfo_t ),
+                                   globalUnsubscribePacketIdentifier );
+
+    if( mqttStatus != MQTTSuccess )
+    {
+        LogError( ( "Failed to send UNSUBSCRIBE packet to broker with error = %u.",
+                    mqttStatus ) );
+        returnStatus = EXIT_FAILURE;
+    }
+    else
+    {
+        LogInfo( ( "UNSUBSCRIBE sent for topic %.*s to broker.\n\n",
+                   topicFilterLength,
+                   pTopicFilter ) );
+
+        /* Process Incoming UNSUBACK packet from the broker. */
+        mqttStatus = MQTT_ProcessLoop( pMqttContext, MQTT_PROCESS_LOOP_TIMEOUT_MS );
+
+        if( mqttStatus != MQTTSuccess )
+        {
+            returnStatus = EXIT_FAILURE;
+            LogError( ( "MQTT_ProcessLoop failed: Status=%s.",
+                        MQTT_Status_strerror( mqttStatus ) ) );
+        }
+    }
+
+    return returnStatus;
+}
+
+/*-----------------------------------------------------------*/
+
+static int publishToTopic( MQTTContext_t * pMqttContext,
```

Comment:
This function is only used for sending publishes to broker, and not for receiving incoming PUBLISHes

Reply:
On line 1186, you use this function for receiving incoming publishes with the call to MQTT_ProcessLoop(). Otherwise, the check on line 1283 for receiving the temperature data would fail.

Reply:
Ahh, okay good point! Have renamed the function to `publishToTopicAndProcessIncomingMessage`